### PR TITLE
Adjustments to data table expansion logic

### PR DESCRIPTION
**Pull request by DrewHoo**
#### **Verify that...**

- [x] Any related demos were added and `npm start` and `npm run build` still works
- [x] New demos work in `Safari`, `Chrome` and `Firefox`
- [x] `npm run lint` passes
- [x] `npm test` passes and code coverage is increased
- [x] `npm run build` still works

#### **Bullhorn Internal Developers**
- [x] Run `Novo Automation`
- [ ] Run `BBO Automation`

##### **Screenshots**

### CODE REVIEW

File: projects/novo-elements/src/elements/data-table/data-table-expand.directive.ts [hunk 1]
```diff
@@ -17,11 +17,13 @@ export class NovoDataTableExpandDirective<T> implements OnDestroy {
...
 
   constructor(public vcRef: ViewContainerRef, private state: DataTableState<T>, private dataTable: NovoDataTable<T>) {
-    this.subscription = this.state.expandSource.subscribe(() => {
-      if (dataTable.isExpanded(this.row)) {
-        this.render();
-      } else {
-        this.clear();
+    this.subscription = this.state.expandSource.subscribe((targetId?: number) => {
+      if ([undefined, ((this.row as unknown) as { id: number }).id].includes(targetId)) {
+        if (dataTable.isExpanded(this.row)) {
```
Comment (amrutha-rajiv) [inline projects/novo-elements/src/elements/data-table/data-table-expand.directive.ts]: why is undefined explicitly called out here?

Comment (DrewHoo) [reply projects/novo-elements/src/elements/data-table/data-table-expand.directive.ts]: @chris-erxleben @amrutha-rajiv I pushed some changes to make this more straightforward. LMK what you think and resolve if you think it's suitable.

**Comment by DrewHoo** [comment]
I reran sanities after that last small change: http://automation1.bh-bos2.bullhorn.com/job/Development/job/Novo/job/Automation/job/Sanity/job/Sanity%20-%20QA/710/

File: projects/novo-elements/src/elements/data-table/data-table-expand.directive.ts [hunk 2]
```diff
@@ -17,15 +17,21 @@ export class NovoDataTableExpandDirective<T> implements OnDestroy {
...
+          this.render();
+        } else {
+          this.clear();
+        }
       }
     });
   }
 
+  shouldExpandAllRows = (targetId: number): boolean => targetId === undefined;
```
Comment (chris-erxleben) [inline projects/novo-elements/src/elements/data-table/data-table-expand.directive.ts]: I think these would look better as getters

**Pull request merged**